agent/engine/metadataservice/types.go

```diff
@@ -0,0 +1,47 @@
+package metadataservice
+
+type Metadata struct {
```

Comment:
Please use `gofmt` to format the code before save. You can use this [tool](https://github.com/fatih/vim-go-tutorial) for convenience.